src/Mvc/Mvc.ViewFeatures/src/RazorComponents/ComponentRenderer.cs

```diff
@@ -147,4 +191,22 @@ private IHtmlContent NonPrerenderedWebAssemblyComponent(HttpContext context, Typ
             return new ComponentHtmlContent(_WebAssemblyComponentSerializer.GetPreamble(currentInvocation));
         }
     }
+
+    internal class InvokedRenderModes
```

Comment:
Is there a reason you need this indirection? (Could it just be the enum)?

Reply:
I wanted to avoid the continuous boxing/unboxing in/out of the Items collection. Am I missing something here?